### PR TITLE
Move wrappers to new module

### PR DESCRIPTION
**Pull request by manicminer**
This is an attempt to move the wrappers out of Hamilton that pull in a dependency on go-autorest. This makes it more palatable to consume, and lays some ground for future library consolidation and weaning off go-autorest.

it also introduces a `EnvironmentFromAzureEnvironment()` function which converts between `azure.Environment` and `environments.Environment`. This avoids interim duplication of logic that interacts with the Azure Metadata Service and should enable compatibility for custom environments to be restored to the AzureRM TF Provider.

### CODE REVIEW

**Review by katbyte — APPROVED** [review]
LGTM 🦀

**Comment by manicminer** [comment]
Test results

Screenshot 2022-03-10 at 19 27 49

File: auth/cache.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/manicminer/hamilton/auth"
+)
+
+type CachedAutorestAuthorizer struct {
```
Comment (tombuildsstuff) [inline auth/cache.go]: how is this cached? should this be HamiltonAutorestAuthorizer?

**Review by tombuildsstuff — APPROVED** [review]
One comment around naming but this otherwise LGTM 👍

File: auth/cache.go (hunk 1 above)
Comment (manicminer) [reply auth/cache.go]: Good catch thanks, it isn't doing any caching (that functionality remains in the origin struct in the main hamilton auth package)

Comment (manicminer) [reply auth/cache.go]: Going to rename all the types to drop the Autorest prefix too since they now live in their own package

**Pull request merged**